packages/jest-runtime/src/index.js

```diff
@@ -412,10 +414,51 @@ class Runtime {
   }
 
   requireModuleOrMock(from: Path, moduleName: string) {
-    if (this._shouldMock(from, moduleName)) {
-      return this.requireMock(from, moduleName);
-    } else {
-      return this.requireModule(from, moduleName);
+    try {
+      if (this._shouldMock(from, moduleName)) {
+        return this.requireMock(from, moduleName);
+      } else {
+        return this.requireModule(from, moduleName);
+      }
+    } catch (e) {
+      if (e.code === 'MODULE_NOT_FOUND') {
+        if (!path.isAbsolute(moduleName) && path.extname(moduleName) === '') {
+          const dirname = path.dirname(from);
+          const pathToModule = path.resolve(dirname, moduleName);
+
+          try {
+            const slashedDirname = slash(dirname);
+
+            const matches = glob(`${pathToModule}.*`)
```

Comment:
would be nice to have the hastemap here, but unless I'm missing something it's not available from `runtime`